v0/destinations/singular/transform.js

```diff
@@ -0,0 +1,215 @@
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  BASE_URL,
+  sessionEvents,
+  SINGULAR_SESSION_ANDROID_EXCLUSION,
+  SINGULAR_SESSION_IOS_EXCLUSION,
+  SINGULAR_EVENT_ANDROID_EXCLUSION,
+  SINGULAR_EVENT_IOS_EXCLUSION
+} = require("./config");
+const {
+  defaultRequestConfig,
+  constructPayload,
+  CustomError,
+  defaultGetRequestConfig,
+  removeUndefinedAndNullAndEmptyValues,
+  getSuccessRespEvents,
+  getErrorRespEvents,
+  extractCustomFields,
+  getValueFromMessage
+} = require("../../util");
+
+const responseBuilderSimple = (message, { Config }) => {
+  // trim and replace spaces with "_"
+  message.event = message.event.trim().replace(/\s+/g, " ");
+  const eventType = message.event;
+  const response = defaultRequestConfig();
+  const eventAttributes = {};
+  let payload, endPoint;
+  const platform = getValueFromMessage(message, "context.os.name");
+  if (
+    Config.sessionEventList.includes(eventType) ||
+    sessionEvents.includes(eventType)
+  ) {
+    if (platform.toLowerCase() === "android") {
+      payload = constructPayload(
+        message,
+        MAPPING_CONFIG[CONFIG_CATEGORIES.SESSION_ANDROID.name]
+      );
+      if (!payload) {
+        // fail-safety for developer error
+        throw new CustomError("Failed to Create Android Session Payload", 400);
+      }
+      extractCustomFields(
+        message,
+        eventAttributes,
+        ["properties"],
+        SINGULAR_SESSION_ANDROID_EXCLUSION
+      );
+    } else if (platform.toLowerCase() === "ios") {
+      payload = constructPayload(
+        message,
+        MAPPING_CONFIG[CONFIG_CATEGORIES.SESSION_IOS.name]
+      );
+      if (!payload) {
+        // fail-safety for developer error
+        throw new CustomError("Failed to Create iOS Session Payload", 400);
+      }
+      extractCustomFields(
+        message,
+        eventAttributes,
+        ["properties"],
+        SINGULAR_SESSION_IOS_EXCLUSION
+      );
+
+      /*
+        if att_authorization_status is true then dnt will be false,
+        else by default dnt value is true
+      */
+      payload.dnt = !payload.att_authorization_status;
+    } else {
+      throw new CustomError("[Singular] :: Invalid Platform", 400);
+    }
```

Comment:
This portion is repetitive for both the session events and other events. Define a function and use that.

Reply:
We are having separate exclusion lists depending on the platform as well as the type of event. This was done to avoid passing the exclusion lists alongside.

Reply:
Yeah but this portion is the same line by line so we should remove it and create one function for that.

Reply:
done